e2e/cypress/integration/post_header/post_header_spec.js

```diff
@@ -217,4 +235,38 @@ describe('Post Header', () => {
             cy.get(`#SEARCH_flagIcon_${postId}`).siblings('.post__pinned-badge').should('exist');
         });
     });
+
+    it('M17442 Visual verification of "Searching" animation for Flagged and Pinned posts', () => {
+        // # Go to Town-Square channel
+        cy.visit('/ad-1/channels/town-square');
+
+        // Pin and flag ten posts before clicking on Pinned and Flagged post icons
+        pinAndFlagTenPosts();
```

Comment:
Please change above lines to:
```javascript
        cy.delayRequestToRoutes(['pinned', 'flagged'], 5000);
        cy.reload();

        // # Go to Town-Square channel
        cy.visit('/ad-1/channels/town-square');

        // Pin and flag ten posts before clicking on Pinned and Flagged post icons
        cy.postMessage('Post');

        //Pin and flag the posted message
        cy.getLastPostId().then((postId) => {
            cy.clickPostDotMenu(postId);
            cy.get(`#pin_post_${postId}`).click();
            cy.clickPostFlagIcon(postId);
        });
```

Then add this new Cypress command, preferably in the new file `cypress/support/fetch_commands.js`
```javascript
Cypress.Commands.add('delayRequestToRoutes', (routes = [], delay = 0) => {
    cy.on('window:before:load', (win) => addDelay(win, routes, delay));
});

const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

const addDelay = (win, routes, delay) => {
    const fetch = win.fetch;
    cy.stub(win, 'fetch').callsFake((...args) => {
        for (let i = 0; i < routes.length; i++) {
            if (args[0].includes(routes[i])) {
                return wait(delay).then(() => fetch(...args));
            }
        }

        return fetch(...args);
    });
};
```

Be sure to import this new file at `e2e/cypress/support/index.js`

Reply:
Updated and added fetch_commands.js